src/panels/lovelace/mixins/element-click-mixin.js

```diff
@@ -12,8 +16,68 @@ import computeStateName from "../../../common/entity/compute_state_name";
 export default dedupingMixin(
   (superClass) =>
     class extends NavigateMixin(EventsMixin(superClass)) {
-      handleClick(hass, config) {
-        const tapAction = config.tap_action || "more-info";
+      registerMouse(config) {
+        const indicator = this.shadowRoot.querySelector("div mwc-ripple");
```

Comment:
I've realized, we should not patch every element to have a ripple but we should create a ripple when we need it and inject it into the shadow dom.

```js
let ripple = null;

const startAnimation = () => {
  if (ripple === null) {
    ripple = document.createElement('mwc-ripple');
    this.shadowRoot.append(ripple);
  }
  // …
}
```

Reply:
That way we're also not going to render 100 ripples on startup and realize we only need 1 when a user clicks on it.

Reply:
relevant talk: https://youtu.be/zfQoleQEa4w?t=1373

Reply:
(it actually talks about ripples! 😆)

Reply:
Unfortunately, ripple doesn't seem to work if placed directly in the shadowroot.

Reply:
I can probably get it to work if all elements have a wrapping div, though.

Reply:
Fixed. Also looks better when the ripple doesn't happen on short presses too, imo.